GitUI/CommandsDialogs/FormBrowse.cs

```diff
@@ -1662,30 +1649,32 @@ private void ChangeWorkingDir(string path)
                 }
             }
 
-            bool removedAll = false;
-            if (invalidPaths.Count > 1)
-            {
-                dialogResult = MessageBox.Show(this, string.Format(_directoriesAreNotValidRepositoriesFormat.Text, invalidPaths.Count),
-                                                            _directoriesAreNotValidRepositoriesCaption.Text,
-                                                            MessageBoxButtons.YesNo,
-                                                            MessageBoxIcon.Exclamation,
-                                                            MessageBoxDefaultButton.Button1);
+            string dialogTitle = _directoryIsNotAValidRepositoryCaption.Text;
+            string mainInstruction = _directoryIsNotAValidRepositoryMainInstruction.Text;
+            string content = "";
+            string commandButtons = Smart.Format(_directoryIsNotAValidRepositoryCommandButtons.Text, invalidPaths.Count > 1, invalidPaths.Count, '|');
+            const bool showCancelButton = true;
 
-                if (dialogResult == DialogResult.Yes)
-                {
-                    foreach (var invalidPath in invalidPaths)
-                    {
-                        ThreadHelper.JoinableTaskFactory.Run(() => RepositoryHistoryManager.Locals.RemoveRecentAsync(invalidPath));
-                    }
+            int dialogResult = PSTaskDialog.cTaskDialog.ShowCommandBox(dialogTitle, mainInstruction, content, commandButtons, showCancelButton);
 
-                    removedAll = true;
-                }
+            if (dialogResult < 0)
+            {
+                /* Cancel */
+                return;
             }
-
-            if (!removedAll)
+            else if (PSTaskDialog.cTaskDialog.CommandButtonResult == 0)
             {
+                /* Remove selected invalid repo */
                 ThreadHelper.JoinableTaskFactory.Run(() => RepositoryHistoryManager.Locals.RemoveRecentAsync(path));
             }
+            else if (PSTaskDialog.cTaskDialog.CommandButtonResult == 1)
+            {
+                /* Remove all invalid repos */
+                foreach (var invalidPath in invalidPaths)
+                {
+                    ThreadHelper.JoinableTaskFactory.Run(() => RepositoryHistoryManager.Locals.RemoveRecentAsync(invalidPath));
+                }
```

Comment:
Please use `RemoveInvalidRepositoriesAsync` instead.

Deleting each individual repo in current fashion is quite inefficient, as it will be opening and closing the underlying config for each deletion.
It may also introduce racing conditions - multiple threads would attempt to modify the same file. It